File: golang/368_Largest_Divisible_Subset.go
package main

import (
	"sort"
)

func largestDivisibleSubset(nums []int) []int {
	S := make(map[int][]int)
	S[-1] = []int{}
	sort.Ints(nums)
	for _, value := range nums {
		max := []int{}
		for k, v := range S {
			if value%k == 0 && len(v) > len(max) {
				max = v
			}
		}
		max = append(max, value)
		S[value] = max
	}
	m := []int{}
	for _, value := range S {
		if len(value) > len(m) {
			m = value
		}
	}
	return m
}

File: Python/368_Largest_Divisible_Subset.py
class Solution:
    def largestDivisibleSubset(self, nums: List[int]) -> List[int]:
        S = {-1: []}
        import copy
        for x in sorted(nums):
            S[x] = copy.copy(max((S[d] for d in S if x % d == 0), key=len)) +[x]
        return max(S.values(), key=len)
